web/services/sap_systems.go

```diff
@@ -50,6 +51,24 @@ func (s *sapSystemsService) GetAllApplications(filter *SAPSystemFilter, page *Pa
 func (s *sapSystemsService) GetAllDatabases(filter *SAPSystemFilter, page *Page) (models.SAPSystemList, error) {
 	return s.getAllByType(models.SAPSystemTypeDatabase, models.TagDatabaseResourceType, filter, page)
 }
+func (s *sapSystemsService) GetByID(ID string) (*models.SAPSystem, error) {
+	var instances entities.SAPSystemInstances
+
+	err := s.db.
+		Where("id = ?", ID).
+		Find(&instances).
+		Error
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(instances) == 0 {
+		return nil, nil
+	}
+
+	return instances.ToModel()[0], nil
```

Comment:
Why do we return only 1 item here?
Do we assume that all instances have the same data? (which in some cases is true, but as you are loading here the instance data...)

Reply:
The `toModel` function takes a list of sap system instances (entities), groups them by ID, and returns a list of SAPSystems (models) by aggregating data.
Here since we specified that we want just one ID in the condition above, a list containing only one SAPSystem instance will be returned everytime.